### PR TITLE
Add spaCy tasks

### PR DESCRIPTION
**Pull request by zangell44**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?

This PR adds five new tasks for interaction with spaCy. Tasks include running a spaCy model, and accessing various attributes (Tagger, Parser, Named Entity Recognizer, custom).

Basic tests have been added for all operations. Local testing has also been performed with pre-loaded spaCy models.

## Why is this PR important?

Closes [#1018](https://github.com/PrefectHQ/prefect/issues/1018)

### CODE REVIEW

**Comment by joshmeek** [comment]
Awesome! Testing this out now to see how the workflow feels

File: tests/tasks/spacy/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+import pytest
```
Comment (joshmeek) [inline tests/tasks/spacy/__init__.py]: To pass the vanilla prefect tests do:
```
import pytest

pytest.importorskip("spacy")
```

Comment (zangell44) [reply tests/tasks/spacy/__init__.py]: thanks! I had been searching everywhere for a missing dependency

File: src/prefect/tasks/spacy/spacy_tasks.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        Args:
+            - nlp (spaCy text processing pipeline, optional): a custom spaCy text
+                processing pipeline, must be provided if not
+                specified in construction
+
+        Returns:
+            - Tagger: spaCy Tagger object
+
+        """
+        return nlp.tagger
```
Comment (joshmeek) [inline src/prefect/tasks/spacy/spacy_tasks.py]: I think that tasks should raise a value error if `nlp` is `None` (never passed in during run or initialization)

Comment (zangell44) [reply src/prefect/tasks/spacy/spacy_tasks.py]: Added logic and tests in latest commit

**Comment by zangell44** [comment]
> Looking good! Do you mind writing up a short example in the `/examples` directory?

Put together a quick example, let me know what you think. Is there any references to `/examples` that should be updated for the new file?

**Comment by joshmeek** [comment]
> Put together a quick example, let me know what you think. Is there any references to `/examples` that should be updated for the new file?

Ah yeah, `docs/guide/examples`, should be added there.

**Comment by joshmeek** [comment]
@cicdw Take a pass through, lgtm

File: src/prefect/tasks/spacy/spacy_tasks.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        text: str = u"",
+        nlp=None,
+        spacy_model_name: str = "en_core_web_sm",
+        disable: list = [],
```
Comment (cicdw) [inline src/prefect/tasks/spacy/spacy_tasks.py]: Having mutable default arguments can lead to weird and unexpected behavior; could you default these to `None` and do a simple `self.disable = disable or []` in the body of the method?  Same for `component_cfg`

File: src/prefect/tasks/spacy/spacy_tasks.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+                self.nlp = spacy.load(
+                    spacy_model_name, disable=disable, component_cfg=component_cfg
+                )
+            except IOError:
+                raise ValueError("spaCy model %s not found." % spacy_model_name)
+
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("text")
+    def run(self, text: str = u""):
```
Comment (cicdw) [inline src/prefect/tasks/spacy/spacy_tasks.py]: Any chance you could write a test to verify that `defaults_from_attrs` works the way we expect in this situation?  Specifically, a test which verifies that if a non-trivial `text` was provided at initialization and then nothing is provided at runtime, that the initialized `text` attribute it used?  You could just initialize the class with a `MagicMock` `nlp` attribute and verify that it was called with the correct `text`.

File: src/prefect/tasks/spacy/spacy_tasks.py (hunk 3 above)
Comment (zangell44) [reply src/prefect/tasks/spacy/spacy_tasks.py]: Good catch! Fixed in latest commits

File: src/prefect/tasks/spacy/spacy_tasks.py (hunk 4 above)
Comment (zangell44) [reply src/prefect/tasks/spacy/spacy_tasks.py]: Added test in latest commits

**Comment by cicdw** [comment]
Awesome - thanks again for more great additions to the task library!

**Pull request merged**